src/transformers/modeling_utils.py

```diff
@@ -1525,6 +1532,28 @@ def _autoset_attn_implementation(
             # If a config is passed with a preset attn_implementation, we skip the automatic dispatch and use the user-provided config, with hard checks that the requested attention implementation is available.
             requested_attn_implementation = config._attn_implementation_internal
 
+        # Composite models consisting of several PretrainedModels have to specify attention impl as a dict
+        # where keys are sub-config names. But most people will specify one `str` which means that should dispatch
+        # for all sub-models or do not specify anything (`None`).
+        # Below we check is a models is composite and manually prepare a dict of attn impl if not already passed as a dict.
+        # Later each sub-model will dispatch with its own attn impl, by calling `_from_config(attn_impl="sdpa/FA2/eager")`
+        # If any of sub-models don't support requested attn, an error will be raised
+        sub_configs = {
+            key: getattr(config, key) for key in config if isinstance(getattr(config, key), PretrainedConfig)
+        }
+        if sub_configs:  # so we have a composite model
+            attn_implementation_per_subconfig = {}
+            for key, sub_config in sub_configs.items():
+                attn_implementation_per_subconfig[key] = (
+                    requested_attn_implementation
+                    if not isinstance(requested_attn_implementation, dict)
+                    else requested_attn_implementation.get(key)
+                )
+
+            if cls._is_composite:  # some composite models don't use attn impl, e.g. VQ-VAE
```

Comment:
This is for specific cases when a model is composite (has several standalone models) but not all of the components are a Transformer. So the sub-component doesn't/can/t have any attention mechanism and has nothing to dispatch. 

Example: Chameleon with its VQ-VAE or CLIPText models. 

In case of CLIP, the model is mostly loaded with the general config (not text config). So when a `ClipTextModel.from_pretrained()` is loaded it relies on general `ClipConfig` and we replace the config's attention implementation with a dict for each sub-model (text vs vision config). In later steps we fail to dispatch because we are dispatching on `self.config._attn_implementation` and  `attention implementation should be a string and one of the following....`. Hope this explains why. I thinkg for Chameleon we could find a way without `is_composite` attribute but in case of CLIP-like models that would be harder